python/ray/data/dataset.py

```diff
@@ -4455,76 +4444,15 @@ def repeat(self, times: Optional[int] = None) -> "DatasetPipeline":
             times: The number of times to loop over this dataset, or None
                 to repeat indefinitely.
         """
-        from ray.data._internal.plan import _rewrite_read_stage
-        from ray.data.dataset_pipeline import DatasetPipeline
-
-        ctx = DataContext.get_current()
-        if self._plan.is_read_stage_equivalent() and ctx.optimize_fuse_read_stages:
-            blocks, _, stages = self._plan._get_source_blocks_and_stages()
-            blocks.clear()
-            blocks, outer_stats, stages = _rewrite_read_stage(blocks, stages)
-            read_stage = stages[0]
-        else:
-            blocks = self._plan.execute()
-            outer_stats = self._plan.stats()
-            read_stage = None
-        uuid = self._get_uuid()
-        outer_stats.dataset_uuid = uuid
-
-        if times is not None and times < 1:
-            raise ValueError("`times` must be >= 1, got {}".format(times))
-
-        class Iterator:
-            def __init__(self, blocks):
-                self._blocks = blocks
-                self._i = 0
-
-            def __next__(self) -> Callable[[], "Dataset"]:
-                if times and self._i >= times:
-                    raise StopIteration
-                epoch = self._i
-                blocks = self._blocks
-                self._i += 1
-
-                def gen():
-                    ds = Dataset(
-                        ExecutionPlan(
-                            blocks,
-                            outer_stats,
-                            dataset_uuid=uuid,
-                            run_by_consumer=True,
-                        ),
-                        epoch,
-                        lazy=False,
-                    )
-                    ds._set_uuid(uuid)
-                    return ds
-
-                return gen
-
-        class Iterable:
-            def __init__(self, blocks):
-                self._blocks = blocks
-
-            def __iter__(self):
-                return Iterator(self._blocks)
-
-        pipe = DatasetPipeline(Iterable(blocks), False, length=times or float("inf"))
-        if read_stage:
-            pipe = pipe.foreach_window(
-                lambda ds, read_stage=read_stage: Dataset(
-                    ds._plan.with_stage(read_stage), ds._epoch, True
-                )
-            )
-        return pipe
+        raise DeprecationWarning("`DatasetPipeline` is deprecated.")
```

Comment:
Let's add some more wording for people to avoid any surprise, similar to BatchPredictor (https://github.com/ray-project/ray/blob/master/python/ray/train/batch_predictor.py#L10-L17)

something like:

```
`DatasetPipeline` is deprecated from Ray 2.8.
Use `Dataset` instead. `Dataset` supports lazy and streaming execution natively.
Learn more from https://docs.ray.io/en/latest/data/data-internals.html#execution
```